### ISSUE TITLE
How does CatBost use Newton method to minimize the expected loss?

### ISSUE THREAD

**Issue by Wangpc-972** [catboost/catboost#1844]
Hello CatBoost team：
 
In this [paper](https://arxiv.org/abs/1706.09516), the negative gradient descent method is used to minimize the expected loss. If the Newton method is used, I have a few questions to ask you.

1. Can the [g(x)](https://arxiv.org/abs/1706.09516) of formula 2 in the paper be replaced by the second-order partial derivative [here](https://catboost.ai/docs/concepts/algorithm-score-functions.html#second-order-functions)?
2. It is mentioned [here](https://catboost.ai/docs/concepts/algorithm-score-functions.html#gradient-boosting) that the score function can measure the quality of the gradient. What does the score function measure? How to judge good and bad?
3. If the Newton method is used to minimize the expected loss, should the score function use NewtonCosine or NewtonL2?
 
Looking forward to your reply! Thank you!

**Comment by kizill** [comment]
1) CatBoost can calculate leaf values either stepping in antigradient direction or with calculation leaf values via Newton, so in this formula (if i understand you right) so in this formula we can also use a calculated via second order Newton value.
2) Score function measures not a quality of gradient, but an expected loss improvement after adding a new split to tree. So there are two ways to do that: calculating L2 or cosine distance between leaf value calculated as mean gradient in that leaf or in a [second order scheme](https://catboost.ai/docs/concepts/algorithm-score-functions.html#second-order-functions) and a individual first order derivatives of objects.
Or, simply saying, just a distance between a single object first derivative of loss function and a prediction value for a leaf where object will be after adding some split to model.
3) There is no strong connection between score function and a method to calculate final leaf values for you model, by default it's first order Cosine distance - in most cases it's inpractical to use heavy recalculation for value in leaf especially if you are optimising some function with nondiagonal hessian matrix - the family of pairwise loss functions, such as 
YetiRankPairwise, PairLogitPairwise and others. And also for such a heavy losses we are using Gradient method for final leaf values estimation.

You can look at default leaf estimation methods for different loss functions [here in code](https://github.com/catboost/catboost/blob/215e0a04335079bc1dbffbf56cd2ce56eb239bcc/catboost/private/libs/options/catboost_options.cpp#L29-L256)
